src/main.rs

```diff
@@ -1252,7 +1287,7 @@ mod cli {
 
         /// Scope C code using a custom tree-sitter query.
         #[arg(long, env, verbatim_doc_comment, value_name = TREE_SITTER_QUERY_VALUE_NAME)]
-        pub c_query: Vec<CustomCQuery>,
+        pub c_query: Vec<CodeQuery>,
```

Comment:
So now `CodeQuery` is "less typesafe": all languages share a single type. How do we ensure we do not accidentally pass e.g. a Python tree-sitter query string to something expecting a different language? (that likely fails but MIGHT NOT, at which point confusion ensues)

Reply:
The only place I see where an bug can occur is in the mapping of commands to languages in the `handle_language_scope` macro. And that mapping is simple to verify as correct by reading the mappings.

You can always add a `PhantomData<Lang>` to `CodeQuery` but for me it's not winning much. It feels like typing for the sake of typing.

Do you see some other issue that stronger typing would prevent?

Reply:
Nope, makes sense